Source/WebCore/rendering/RenderTheme.cpp

```diff
@@ -834,8 +838,14 @@ bool RenderTheme::isControlStyled(const RenderStyle& style, const RenderStyle& u
     // FIXME: SearchFieldPart should be included here when making search fields style-able.
     case TextFieldPart:
     case TextAreaPart:
-        // Test the style to see if the UA border and background match.
-        return !style.borderAndBackgroundEqual(userAgentStyle);
+#if PLATFORM(COCOA)
```

Comment:
Rather than add `PLATFORM(COCOA)` to a platform-agnostic file, we should only perform the link check in our platform-specific files (`RenderThemeMac` / `RenderThemeIOS`, or more ideally `RenderThemeCocoa`). 

You can move / adjust / copy the logic as needed.